services/speech/soapbox/connection.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+import websockets
+
+from soapbox.message import Message
+
+
+def connect(host, port):
+    return websockets.connect('ws://' + host + ':' + port)
+
+
+async def connect_and_handle_messages(host, port, handler):
+    websocket = await connect(host, port)
+    print("connected")
+    while True:
+        try:
+            data = await websocket.recv()
+        except websockets.exceptions.ConnectionClosed:
+            print("connection has closed")
+            raise
+
+        message = None
+        response = None
+
+        try:
+            message = Message.from_json(data)
+        except:
+            print("invalid message")
```

Comment:
You might want a `break` here, instead of continuing to check for values that won't exist.

Reply:
I thought `break` would break out of the `while True` loop? If that's the case then the WebSocket would stop processing incoming messages. I'll investigate because that would makek things much cleaner if not.

Reply:
I think I meant `continue` 😳